### ISSUE TITLE
Media type matching with dynamic properties

### ISSUE THREAD

**Issue by lindyhopchris** [neomerx/json-api#221]
Using the media type classes for content negotiation. Given this scenario:

```http
POST /api/v1/users HTTP/1.1
Host: localhost:8000
Accept: application/vnd.api+json
Content-Type: multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW

------WebKitFormBoundary7MA4YWxkTrZu0gW
Content-Disposition: form-data; name="user"; filename="user.xml"
Content-Type: text/xml
------WebKitFormBoundary7MA4YWxkTrZu0gW--
```

It's not possible to match the content type using `MediaType::equalsTo()` or `MediaType::matches()` because both use `MediaType::isMediaParametersEqual()`... for which there's no way to specify that you don't care what the `boundary` parameter value is... i.e. you cannot specify wildcard parameter values.

I'm not quite sure how to implement it, but do you think there's a way of allow wildcard parameter values when comparing the parameters?

**Comment by neomerx** [comment]
OK, I'll try to find something about how to deal with it either in JSON API spec or a web spec.

Btw, I've just pre-released v3 of the library. Support JSON API 1.1, better-designed and performance testing shows improvement 2,3 - 5,4 X. Placed in `develop`.

**Comment by lindyhopchris** [comment]
@neomerx nice! I'm still stuck a little bit behind but hoping to catch up to your latest version sometime soon.

**Comment by neomerx** [comment]
Added support for `*` in parameters to `matchesTo` method in v3. It's now possible

```php
$type1 = new MediaType('multipart', 'form-data', ['boundary' => '*']);
$type2 = new MediaType('multipart', 'form-data', ['boundary' => '----WebKitFormBoundaryAAA']);

self::assertTrue($type2->matchesTo($type1));
```

Full test code [here](https://github.com/neomerx/json-api/blob/e17d3ba6b8c1f7f9f388407e8f2429f7c198959b/tests/Http/Headers/MediaTypeTest.php#L144-L174).

**Comment by lindyhopchris** [comment]
Great! Now I just need to upgrade to v3! Thanks for adding it in.

**Comment by neomerx** [comment]
released as [v3.0.0-rc.2](https://github.com/neomerx/json-api/releases/tag/v3.0.0-rc.2)

@lindyhopchris Don't hesitate to ask migration questions :wink:

**Comment by lindyhopchris** [comment]
will do!

**Comment by neomerx** [comment]
released as [v3.0.0](https://github.com/neomerx/json-api/releases/tag/v3.0.0)
